frontend/src/views/photo/photo.ts

```diff
@@ -81,19 +81,29 @@ export async function listPhotosInAlbum(albumId: string, low_res: boolean = fals
     return result.photos.map(protoPhotoToPhotoModel);
 }
 
+export class TooManyRequests {}
+
 /**
  * Create a photo
  * @param albumId The ID of the album
  * @param photoData The bytes of the photo. May be `PNG` or `JPEG` format.
  * @return `true` on success. `undefined` on failure.
  */
-export async function createPhoto(albumId: string, photoData: Uint8Array): Promise<boolean | undefined> {
+export async function createPhoto(albumId: string, photoData: Uint8Array): Promise<boolean | TooManyRequests | undefined> {
     const result = await Http.post('/api/v1/photo', new CreatePhotoRequest({
         albumId,
         photoData
     }), null);
 
-    return result.ok ? true : undefined;
+    if(result.ok) {
+        return true;
+    }
+
+    if(result.status == 429) {
+        return new TooManyRequests();
```

Comment:
TooManyRequests should be a throwable

Reply:
Fair.

We should probably standardize and improve error handling in general on the frontend. I really dislike the undefined type for this

Reply:
Agreed